src/resolvers/user.ts

```diff
@@ -37,17 +38,25 @@ export default {
      * Register user with provided email
      * @param _obj - parent object (undefined for this resolver)
      * @param email - user email
+     * @param utm - Data form where user went to sign up. Used for analytics purposes
      * @param factories - factories for working with models
      */
     async signUp(
       _obj: undefined,
-      { email }: {email: string},
+      { email, utm }: { email: string; utm?: UserDBScheme['utm'] },
       { factories }: ResolverContextBase
     ): Promise<boolean | string> {
+      // Validate and sanitize UTM parameters
+      if (!validateUtmParams(utm)) {
+        throw new UserInputError('Invalid UTM parameters provided');
+      }
```

Comment:
I think we should create user anyways, by don't save invalid utms. Probably, we need to write error in logs and send to Hawk.